From 9b4f2f7c3577981bc7fe0d58e576ed06a66e86bb Mon Sep 17 00:00:00 2001
From: Yuwei Chen <yuwei.chen@intel.com>
Date: Tue, 8 Sep 2020 14:53:43 +0800
Subject: [PATCH] Test case for Add included FakeTest.i changing - asm file

Description: Add included file FakeTest.i in the TestPkg/FakeEmuSec/Ia32/SwitchRam.asm.

Repo: edk2

Signed-off-by: Yuwei Chen <yuwei.chen@intel.com>
---
 TestPkg/FakeEmuSec/Ia32/SwitchRam.asm | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/TestPkg/FakeEmuSec/Ia32/SwitchRam.asm b/TestPkg/FakeEmuSec/Ia32/SwitchRam.asm
index 78188dae89..f90c569784 100644
--- a/TestPkg/FakeEmuSec/Ia32/SwitchRam.asm
+++ b/TestPkg/FakeEmuSec/Ia32/SwitchRam.asm
@@ -13,6 +13,7 @@
 ;
 ;------------------------------------------------------------------------------
 #include "FakeTest.h"
+#include "FakeTest.i"
 #include "FakeTest.inc"
     .586p
     .model  flat,C
@@ -57,8 +58,8 @@ SecSwitchStack   PROC
     mov   dword ptr [eax + testaddr0], edx
     mov   edx, dword ptr [esp + testaddr1]
     mov   dword ptr [eax + testaddr1], edx
-    mov   edx, dword ptr [esp + 14]    ; Update this function's return address into permanent memory
-    mov   dword ptr [eax + 14], edx
+    mov   edx, dword ptr [esp + testaddr2]    ; Update this function's return address into permanent memory
+    mov   dword ptr [eax + testaddr2], edx
     mov   esp, eax                     ; From now, esp is pointed to permanent memory
 
     ;
-- 
2.27.0.windows.1

